integration/integration_test.go

```diff
@@ -87,26 +87,31 @@ func TestMain(m *testing.M) {
 		createTestNamespace,
 	}
 
+	cleanup := func() {
+		fmt.Printf("Clean inspektor-gadget:\n")
+		cleanupTestNamespace.runWithoutTest()
+		cleanupInspektorGadget.runWithoutTest()
+	}
+
+	// defer the cleanup to be sure it's called if the test
+	// fails (hence calling runtime.Goexit())
+	defer cleanup()
+
 	fmt.Printf("Setup inspektor-gadget:\n")
 	for _, cmd := range initCommands {
 		err := cmd.runWithoutTest()
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "%v\n", err)
-
+			cleanup()
 			os.Exit(-1)
 		}
 	}
 
-	// defer the cleanup to be sure it's called even if the test
-	// fails (hence calling runtime.Goexit())
-	defer func() {
-		fmt.Printf("Clean inspektor-gadget:\n")
-		cleanupTestNamespace.runWithoutTest()
-		cleanupInspektorGadget.runWithoutTest()
-	}()
-
 	ret := m.Run()
 
+	// os.Exit() doesn't call defered functions, hence do the cleanup manually.
```

Comment:
If the problem comes from `os.Exit()`, what about replacing it by something like this:

```golang
if ret != 0 {
  t.Fatal("return code is %v", ret)
}
```

Reply:
I don't think that's going to work. `os.Exit()` sets the return code of the application and I'm not sure `t.Fatal` does the same. Also the `TestMain` documentation recommends to use `m.Run` with `os.Exit()`.

Reply:
`t.Fatal()` would make the application returning something different than 0, I think.
But you are right, the document advices to do something like this `os.Exit(m.Run())`.

You can merge whenever you want, I already rebased my PR on top of yours.